x-pack/plugins/security_solution/common/api/endpoint/actions/response_actions/kill_process/kill_process.gen.ts

```diff
@@ -14,14 +14,30 @@
  *   version: 2023-10-31
  */
 
-import type { z } from '@kbn/zod';
+import { z } from '@kbn/zod';
 
-import { KillOrSuspendActionSchema, SuccessResponse } from '../../../model/schema/common.gen';
+import { SuccessResponse, BaseActionSchema, Pid, EntityId } from '../../../model/schema/common.gen';
+
+export type KillProcessRouteRequestBody = z.infer<typeof KillProcessRouteRequestBody>;
```

Comment:
is there a benefit to using the infer type here? not too familiar with and would love some tips

Reply:
This is a generated file (see the README file for more on it) and it uses this `Zod` library to create types, so I'm not really familiar with it. But that being said - `z.infer<>` is a Zod specific generic and not a Typescript built in utility.